lib/text/vtt_text_parser.js

```diff
@@ -219,6 +222,75 @@ shaka.text.VttTextParser = class {
     return cue;
   }
 
+  /**
+   * Parses a WebVTT styles from the given payload.
+   *
+   * @param {string} payload
+   * @param {!shaka.text.Cue} rootCue
+   */
+  static parseCueStyles(payload, rootCue) {
+    const xmlPayload = '<span>' + payload + '</span>';
+    const element = shaka.util.XmlUtils.parseXmlString(xmlPayload, 'span');
+    if (element) {
+      /** @type {Array.<!shaka.extern.Cue>} */
+      const cues = [];
+      const VttTextParser = shaka.text.VttTextParser;
+      const childNodes = [...element.childNodes];
+      if (childNodes.length == 1) {
+        const childNode = childNodes[0];
+        if (childNode.nodeType == Node.TEXT_NODE ||
+            childNode.nodeType == Node.CDATA_SECTION_NODE) {
+          rootCue.payload = payload;
+          return;
+        }
+      }
+      for (const childElement of childNodes) {
+        VttTextParser.generateCueFromElement_(childElement, rootCue, cues);
+      }
+      rootCue.nestedCues = cues;
+    } else {
+      rootCue.payload = payload;
+    }
+  }
+
+  /**
+   * @param {!Node} element
+   * @param {Array.<!shaka.extern.Cue>} cues
+   * @private
+   */
+  static generateCueFromElement_(element, rootCue, cues) {
+    const nestedCue = rootCue.clone();
+    if (element.nodeType === Node.ELEMENT_NODE && element.nodeName) {
+      const bold = shaka.text.Cue.fontWeight.BOLD;
+      const italic = shaka.text.Cue.fontStyle.ITALIC;
+      const underline = shaka.text.Cue.textDecoration.UNDERLINE;
+      const tags = element.nodeName.split(/[ .]+/).filter(Boolean);
```

Comment:
What does `.filter(Boolean)` accomplish here?  Removing blank strings?  Why?

Reply:
Yes, the idea was to remove the blanks, but looking at it again, it really is not necessary.